include/eve/detail/function/simd/riscv/bitmask.hpp

```diff
@@ -0,0 +1,46 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/module/core/regular/if_else.hpp>
+
+#include <bitset>
+
+namespace eve::detail
+{
+//================================================================================================
+// Logical to Bits
+//================================================================================================
+template<typename T, typename N>
+EVE_FORCEINLINE auto
+to_bits(rvv_ const&, logical<wide<T, N>> p) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  // As return by arm.
+  using int_type       = as_integer_t<T, unsigned>;
+  using u_wide_t       = wide<int_type, N>;
+  using u_mask_t       = logical<u_wide_t>;
+  auto     u_mask      = bit_cast(p, as<u_mask_t> {});
+  u_wide_t if_else_res = if_else(u_mask,
+                                 /* True value */ static_cast<int_type>(-1),
+                                 /* False Value */ static_cast<int_type>(0));
+  return if_else_res;
+}
```

Comment:
What's true on risk-v5? is it not ffff?

Reply:
this doesn't strike me as a good idea.
sve doesn't do top_bits very well.

You want something (ideally an integer) that is all 1111s for true and all 0s for false.

Can risk-v do that

Reply:
This is exactly what is done. I return `wide<T, N>`, where `T` is unsigned with appropriate size, filled with a `1` if corresponding bit of the `logical` was set, otherwise it will have `0`'s.

Representing  `logical` as a `wide<T,N>` with all ones for `true` and all zeroes for `false` will be ineffective for several reasons:
* before a mask operation I will need to cast it to the real mask type, which will be time-consuming
* I will need to use much more vector registers for keeping this logical (at most 8 instead of always 1 for real mask type).